python/ray/serve/pipeline/generate.py

```diff
@@ -0,0 +1,170 @@
+from typing import Any, Dict, List, OrderedDict
+import threading
+
+from ray.experimental.dag import (
+    DAGNode,
+    ClassNode,
+    ClassMethodNode,
+    FunctionNode,
+)
+from ray.serve.api import Deployment, DeploymentConfig
+from ray import serve
+from ray.serve.pipeline.deployment_method_node import DeploymentMethodNode
+from ray.serve.pipeline.deployment_node import DeploymentNode
+
+
+class DeploymentIDGenerator(object):
+    """
+    Generate unique suffix for each given deployment_name requested for id.
+    By default uses deployment_name for the very first time, then append
+    monotonic increasing id to it.
+    """
+
+    __lock = threading.Lock()
+    __shared_state = dict()
+
+    @classmethod
+    def get_deployment_id(cls, deployment_name: str):
+        with cls.__lock:
+            if deployment_name not in cls.__shared_state:
+                cls.__shared_state[deployment_name] = 0
+                return deployment_name
+            else:
+                suffix_num = cls.__shared_state[deployment_name] + 1
+                cls.__shared_state[deployment_name] = suffix_num
```

Comment:
nit: a bit easier to read IMO
```suggestion
                cls.__shared_state[deployment_name] += 1
                suffix_num = cls.__shared_state[deployment_name]
```